datajunction-server/datajunction_server/api/authentication/whoami.py

```diff
@@ -0,0 +1,22 @@
+"""
+Router for getting the current active user
+"""
+from fastapi import APIRouter, Depends, Request
+from fastapi.security import HTTPBearer
+
+from datajunction_server.models.user import UserOutput
+from datajunction_server.utils import get_settings
+
+settings = get_settings()
+router = APIRouter(
+    tags=["Who am I?"],
+    dependencies=[Depends(HTTPBearer())] if settings.secret else [],
+)
```

Comment:
Maybe we should create a `SecureAPIRouter` to remove the need for declaring dependencies everywhere, something like:

```python
class SecureAPIRouter(APIRouter):

    def __init__(self, *args: Any, **kwargs: Any):
        settings = get_settings()
        return super().__init__(
            *args,
            dependencies=[Depends(HTTPBearer())] if settings.secret else [],
            **kwargs,
        )
```

Reply:
Love this. A friend gave me the idea to also wrap HTTPBearer so now we have a `DJHTTPBearer` that's set as part of this `SecureAPIRouter`...very clean!